src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -520,16 +520,30 @@ class C<T> where T : struct
                 // (10,8): warning CS8632: The annotation for nullable reference types should only be used in code within a '[NonNullTypes(true)]' context.
                 //     T? F3;
                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "F3").WithLocation(10, 8),
+                // (10,8): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
+                //     T? F3;
+                Diagnostic(ErrorCode.ERR_NullableUnconstrainedTypeParameter, "F3").WithLocation(10, 8),
                 // (5,5): warning CS8618: Non-nullable field 'F1' is uninitialized.
                 //     A() { }
-                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "A").WithArguments("field", "F1").WithLocation(5, 5));
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "A").WithArguments("field", "F1").WithLocation(5, 5),
+                // (11,5): warning CS8618: Non-nullable field 'F3' is uninitialized.
```

Comment:
> Why is F3 considered non-nullable?

This is an error case, a '?' modifier cannot be applied to its type. I am assuming it's type is still treated as unconstrained and therefore considered as something that could be non-nullable. I can add a PROTOTYPE comment if you think we should investigate this scenario in more details.

---
In reply to: [213853354](https://github.com/dotnet/roslyn/pull/29586#discussion_r213853354) [](ancestors = 213853354)

Reply:
There's a warning on `T? F3` that `?` should only be used in `[NonNullTypes(true)]`, but the meaning of `?` seems unambiguous since `T` is constrained to be a reference type. This second warning seems confusing.

---
In reply to: [213855521](https://github.com/dotnet/roslyn/pull/29586#discussion_r213855521) [](ancestors = 213855521,213853354)

Reply:
>

My guess is that due to the error below, we end up either dropping the '?', or end up with an error type symbol for F3. 

                // (10,8): error CS8627: A nullable type parameter must be known to be a value type or non-nullable reference type. Consider adding a 'class', 'struct', or type constraint.
                //     T? F3;

I will add a PROTOTYPE comment to get to the bottom of this behavior.



---
In reply to: [213855521](https://github.com/dotnet/roslyn/pull/29586#discussion_r213855521) [](ancestors = 213855521,213853354)